Source/WebCore/rendering/updating/RenderTreeUpdater.cpp

```diff
@@ -760,25 +761,35 @@ static std::optional<DidRepaintAndMarkContainingBlock> repaintAndMarkContainingB
     }
 
     for (auto it = composedTreeDescendantsIterator.begin(), end = composedTreeDescendantsIterator.end(); it != end; ++it) {
+
+        auto renderer = it->renderer();
+        if (!renderer)
+            continue;
+
+        // If child is the start or end of the selection, then clear the selection to
+        // avoid problems of invalid pointers.
+        if (renderer->isSelectionBorder())
+            renderer->frame().selection().setNeedsSelectionUpdate();
+
         auto* element = dynamicDowncast<Element>(*it);
-        if (!element || !element->renderer())
+        if (!element)
             continue;
-        auto& renderer = *element->renderer();
+        auto& renderElement = *element->renderer();
```

Comment:
Instead of getting the renderer out of the element again, we should just `dynamicDowncast<RenderElement>(renderer)`.